keps/sig-instrumentation/0034-distributed-tracing-kep.md

```diff
@@ -0,0 +1,299 @@
+       resources (e.g. update object X based on changes of object Y)
+     - periodic API calls to reconcile state (e.g. periodic fetching state,
+       heartbeats, leader election, etc.)
+  - Will enabling / using the feature result in supporting new API types? **No**
+    How many objects of that type will be supported (and how that translates
+    to limitations for users)?
+  - Will enabling / using the feature result in increasing size or count
+    of the existing API objects?  **Yes.  It adds an annotation to traced objects.**
+  - Will enabling / using the feature result in increasing time taken
+    by any operations covered by [existing SLIs/SLOs][] (e.g. by adding
+    additional work, introducing new steps in between, etc.)? **No**
+    Please describe the details if so.
+  - Will enabling / using the feature result in non-negligible increase
+    of resource usage (CPU, RAM, disk IO, ...) in any components?
+    Things to keep in mind include: additional in-memory state, additional
+    non-trivial computations, excessive access to disks (including increased
+    log volume), significant amount of data sent and/or received over
+    network, etc. Think through this in both small and large cases, again
+    with respect to the [supported limits][].  **The tracing client library has an in-memory cache for outgoing spans.  I believe this is limited to 1Mb by default.  This overhead would apply to all controllers that export spans.**
+* Rollout, Upgrade, and Rollback Planning
+* Dependencies
+  - Does this feature depend on any specific services running in the cluster
+    (e.g., a metrics service)? **Yes.  In the current version of the proposal, users must run the [opencensus agent](https://opencensus.io/service/components/agent/) to configure which backend (e.g. jager, zipkin, etc.) they want telemetry sent to.**
+  - How does this feature respond to complete failures of the services on
+    which it depends?  **Traces will stop being exported, and components will store spans in memory until the buffer is full.**
+  - How does this feature respond to degraded performance or high error rates
+    from services on which it depends? **If the bi-directional grpc streaming connection to the agent cannot be established or is broken, the controller retries the connection every 5 minutes (by default).**
+* Monitoring requirements
+  - How can an operator determine if the feature is in use by workloads?  **The agent produces metrics about telemetry it sends to the backend. 
+ The operator presumably can also see the telemetry being sent on behalf of user workloads.**
+  - How can an operator determine if the feature is functioning properly?  **Same as previous.  The agent has metrics about telemetry that is sent to backends.**
+  - What are the service level indicators an operator can use to determine the
+    health of the service?  **Error rate of sending traces in controllers and opencensus agent.**
```

Comment:
So to clarify - the plan is to expose those metrics from some client library linked into k8s components too?

Reply:
I tried to dig into the opentelemetry-go code, but couldn't find what I was looking for.  I thought I remember there being metrics like that, but I can't find them...

@bogdandrutu do you know if the OpenTelemetry golang client library or exporters emits metrics about the traces they send?  E.g. cumulative number of spans sent or cumulative number of spans dropped?